JOSESwift/Sources/CryptoImplementation/EC.swift

```diff
@@ -175,25 +175,32 @@ internal struct EC {
     /// - Returns: The signature.
     /// - Throws: `ECError` if any errors occur while signing the input data.
     static func sign(_ signingInput: Data, with privateKey: KeyType, and algorithm: SignatureAlgorithm) throws -> Data {
-        // Sign the input as raw elliptic curve coordinates using a hashing algorithm and a private key.
         guard let curveType = algorithm.curveType else {
             throw ECError.invalidCurveDigestAlgorithm
         }
-
-        let digest = try algorithm.createDigest(input: signingInput)
-        var signatureLength = curveType.signatureOctetLength
-
-        guard let signature = NSMutableData(length: signatureLength) else {
-            throw ECError.couldNotAllocateMemoryForSignature
+        guard let secKeyAlgorithm = algorithm.secKeyAlgorithm else {
+            throw ECError.algorithmNotSupported
         }
 
-        let signatureBytes = signature.mutableBytes.assumingMemoryBound(to: UInt8.self)
-        let status = SecKeyRawSign(privateKey, .sigRaw, digest, digest.count, signatureBytes, &signatureLength)
-        if status != errSecSuccess {
-            throw ECError.signingFailed(description: "Error creating signature. (OSStatus: \(status))")
+        var cfErrorRef: Unmanaged<CFError>?
+        let signature = SecKeyCreateSignature(privateKey, secKeyAlgorithm, signingInput as CFData, &cfErrorRef)
+        if cfErrorRef != nil {
+            throw ECError.signingFailed(description: "Error creating signature. (CFError: \(cfErrorRef!.takeRetainedValue()))")
         }
 
-        return signature as Data
+        // unpack BER encoded ASN.1 format signature to raw format as specified for JWS
+        let ecSignatureTLV = [UInt8](signature! as Data)
```

Comment:
Is there a preferable alternative to the force unwrap here?
The signature should always be non-null at this point since the contract of `SecKeyCreateSignature` only allow `nil` here on failure.
If not I would add a comment and disable the linter here.

Reply:
Found another way to fix address this with an `if let` inside a `guard let` and a `fatalError` if they mismatch.